TechnicalDocuments/TechnicalDebt.md

```diff
@@ -0,0 +1,115 @@
+Inventory of Identified Technical Debt and known Legacy Code.
+=============================================================
+
+Technical debt and legacy code has a sad tendency to build up in any software project. As the iOS project has grown both in terms of team members and supported apps we have found that managing technical has become more difficult. Legacy code should be ideally be refactored to clear technical debt when changes are needed. Existing technical debt that should be addressed when working on a new feature should be included in the planning. There is an obvious risk that legacy code is maintained instead of refactored if it is not clear what the technical debt currently is.
+
+## Objective-C code and GatewayManager.
+
+GatewayManager supplies global dependencies to legacy Objective-C code. There was plenty of code in version one and two of the Babylon app that relied heavily on singletons.
+
+| Objective-C view Controllers | Location | Comments |
+| ---------------------------- | ---------| -------- |
+| BillingInformationViewController | Clinical Records | |
+| EditMembershipCodeViewController | Clinical Records | |
+| MembershipTableViewController | Me Tab | |
+| SurveyViewController | Clinical Records | |
+| InsuranceViewController | Clinical Records | |
+| BBAddAdditionalPatientInformationViewControllerV2 | Clinical Records | Might be dead code |
```

Comment:
> Might be dead code

`BBAddAdditionalPatientInformationViewControllerV2` is now only used for date of birth input via `BaseModel`. Screens: `EditMembershipCodeViewController` and `SurveyViewController`. If those are still in use, then first step could be to refactor `BBAddAdditionalPatientInformationViewControllerV2` to be used just for date of birth input. Cases other than `DateOfBirth` can be removed along with related code (e.g. `HeightAndWeightConverter`).

Reply:
Also related to this (couldn't find a fitting section):
There was a transition period when two versions of Additional Info Bento existed. `AdditionalInfoViewModel`, which is no longer in use, along with related types can be removed. After this `AdditionalInfoViewModelV2` can be renamed to `AdditionalInfoViewModel`.

Reply:
Thanks for this. I think we can replace all remaining Objective-C view controllers in one go. If that is not feasible this background information will be very helpful.